spanner/batch.go

```diff
@@ -291,10 +318,23 @@ func (t *BatchReadOnlyTransaction) Execute(ctx context.Context, p *Partition) *R
 				RequestOptions: p.rreq.RequestOptions,
 				ResumeToken:    resumeToken,
 			})
+			if err != nil {
+				return client, err
+			}
+			md, err := client.Header()
+			if err != nil {
+				return nil, err
+			}
```

Comment:
nit: this early return is no longer needed now that we don't return an error if getting the latency stats fails

Reply:
I think this is still needed as there might be an error in getting the Header, and in that case we return the error

Reply:
Yes, but that happens on line 333. The reason that we added the early return above in the first place, was that we would return the error that might be returned `createContextAndCaptureGFELatencyMetrics`. As we don't do that anymore, we can safely wait until line 333 before returning the error from the `client.Header()` method.

Reply:
But then won't md, err := client.Header() override the value of the original error. I will have to add a separate check as md, errGFE : = client.Header()
			if errGFE != nil {
				return nil, errGFE
			}

Reply:
I'm not sure we mean the same return in this case. I think you can safely return the `if err != nil` block starting at line 325. The `err` on line 329 will not override the `err` from the call to `client.Header()`, as that is a different variable, although it does have the same name. That `err` is only visible in the statement on line 329.

So to summarize:
1. The `if err != nil` on line 321 should remain.
2. The `if err != nil` on line 325 can be removed. (And yes, the `md, err := client.Header()` does override the previous value of `err`, but that is not a problem as we know that `err` at that point is `nil`)
3. The `if err := createContextAndCaptureGFELatencyMetrics` on line 329 can safely remain. That will not override anything, as it is a separate variable only defined on line 329.
4. The `return client, err` on line 333 will therefore always return the value of `err` that was returned by `client.Header()`.

Reply:
Okay makes sense. My bad. I thought you meant remove the error return on line 321. Will do that.